tests/python/gaia-ui-tests/gaiatest/apps/base.py

```diff
@@ -128,7 +128,28 @@ def wait_to_not_be_displayed(self):
         Wait(self.marionette).until(lambda m: self.apps.displayed_app.manifest_url != self.manifest_url or
                                               self.apps.displayed_app.entry_point != self.entry_point)
 
+
 class PageRegion(Base):
     def __init__(self, marionette, element):
         self.root_element = element
         Base.__init__(self, marionette)
+
+
+class PageWithHeader(PageRegion):
+
+    def __init__(self, marionette, _header_locator):
+        Base.__init__(self, marionette)
+        self._header_locator = _header_locator
+
+    def go_back(self):
+        self._header.go_back()
```

Comment:
This function is supposed to return a page instance and explicit tell caller which page will be displayed then after.